src/main/java/fr/xephi/authme/process/join/AsynchronousJoin.java

```diff
@@ -179,8 +179,9 @@ private void processJoinSync(Player player, boolean isAuthAvailable, Location lo
 
     private boolean canResumeSession(Player player) {
```

Comment:
What would you think about introducing a `SessionService` that would encapsulate the session handling logic? This has become a large method that I don't really feel is a good fit inside `AsynchronousJoin`.